### DISCUSSION TITLE
Qual é a diferença entre uma função síncrona e uma função assíncrona em JavaScript?

### DISCUSSION THREAD

**Discussion by oeduardostein** [onicolasdelfino/onicolasdelfino#12] [Q&A] [answered]
Qual é a diferença entre uma função síncrona e uma função assíncrona em JavaScript?

**Comment by onicolasdelfino** [comment, marked as answer]
Em JavaScript, uma função síncrona é aquela que é executada de forma sequencial, ou seja, cada instrução espera a anterior ser concluída para ser executada. O fluxo de execução segue uma linha reta e não avança até que o código anterior termine. Por exemplo:

javascript
Copiar código
function funcaoSincrona() {
    console.log('Início');
    console.log('Fim');
}

funcaoSincrona();
// Saída: Início
//         Fim
Já uma função assíncrona permite que partes do código sejam executadas de forma "não bloqueante". Isso significa que outras instruções podem ser processadas enquanto uma operação (como uma requisição de rede) está sendo concluída em segundo plano. Funções assíncronas geralmente usam promises ou async/await:

javascript
Copiar código
async function funcaoAssincrona() {
    console.log('Início');
    await new Promise(resolve => setTimeout(resolve, 2000));  // Simula uma operação assíncrona
    console.log('Fim');
}

funcaoAssincrona();
// Saída: Início
// (Aguarda 2 segundos)
//         Fim
No exemplo assíncrono, a execução não é bloqueada enquanto a operação de espera ocorre.
